### PR TITLE
feat(staking-pool): add partial withdraw

### PR DESCRIPTION
**Pull request by hejkerooo**

### CODE REVIEW

File: e2e/staking-pool.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+        const currentStake = await pool.getStake();
+        expect(currentStake.amount).toStrictEqual(oneEWT.mul(2));
+
+        await pool.withdraw();
+        const stakeAfterWithdraw = await pool.getStake();
+
+        expect(stakeAfterWithdraw.amount).toStrictEqual(zeroEwt);
+
+        const { status } = await pool.getStake();
```
Comment (JGiter) [inline e2e/staking-pool.e2e.ts:170]:
```suggestion
        const { status } = stakeAfterWithdraw;
```

File: e2e/staking-pool.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        expect(status).toBe(StakeStatus.NONSTAKING);
+    });
+
+    it("should withdraw without reward", async () => {
```
Comment (JGiter) [inline e2e/staking-pool.e2e.ts:174]: If this should test withdraw is possible when not staking, then there is no need to put stake before withdraw

**Review by Harasz — APPROVED** [review]

File: e2e/staking-pool.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        expect(poolContributionLimit.eq(contributionLimit)).toEqual(true);
+    });
+
+    it("should be able to unstake", async () => {
+        const pool = await stakingPoolService.getPool();
+
+        await putStake(pool, oneEWT.mul(2));
+
+        await pool.partialWithdraw(oneEWT);
+        await provider.send("evm_mine", []);
```
Comment (JGiter) [inline e2e/staking-pool.e2e.ts:214]: Could you please explain what is this for?

File: src/modules/staking/staking-pool.service.ts [hunk 4]
```diff
@@ -69,6 +77,18 @@ export class StakingPoolService {
...
+    /**
+     * @description
+     * @param value
+     */
+    async partialWithdraw(value: BigNumber | number): Promise<ContractReceipt> {
+        value = BigNumber.from(value);
+
+        const transaction = await this.pool.connect(this.signerService.signer).unstake(value);
+
+        return transaction.wait();
```
Comment (JGiter) [inline src/modules/staking/staking-pool.service.ts:89]: Why this returns receipt? Is it used somehow?

**Pull request merged**